File: app/src/main/java/ca/judacribz/gainzassist/models/ExerciseDao.java
package ca.judacribz.gainzassist.models;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.*;

import java.util.List;

@Dao
public interface ExerciseDao {

    @Insert
    Long insert(Exercise exercise);


    @Query("SELECT * from exercises WHERE workout_id = :workoutId")
    LiveData<List<Exercise>> getLiveFromWorkout(long workoutId);

    @Query("SELECT * from exercises WHERE workout_id = :workoutId")
    List<Exercise> getFromWorkout(long workoutId);

    @Query("SELECT * from  exercises WHERE id = :id")
    LiveData<Exercise> get(long id);

    @Query("SELECT DISTINCT name FROM exercises")
    LiveData<List<String>> getAllUniqueNames();


    @Update
    void update(Exercise... exercise);


    @Delete
    void delete(Exercise... exercises);
}

File: app/src/main/java/ca/gainzassist/data/local/database/ExerciseDao.kt
package ca.gainzassist.data.local.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import ca.gainzassist.domain.model.Exercise

@Dao
interface ExerciseDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(exercise: Exercise): Long

    @Query("SELECT * from exercises WHERE workout_id = :workoutId")
    fun getLiveFromWorkout(workoutId: Long): LiveData<List<Exercise>>

    @Query("SELECT * from exercises WHERE workout_id = :workoutId ORDER BY exercise_number")
    fun getFromWorkout(workoutId: Long): List<Exercise>

    @Query("SELECT * from exercises WHERE id = :id")
    fun getLiveExerciseFromWorkout(id: Long): LiveData<Exercise>

    @Query("SELECT * from  exercises WHERE id = :id")
    fun getLive(id: Long): LiveData<Exercise>

    @Query("SELECT * from  exercises WHERE id = :id")
    fun getName(id: Long): Exercise?

    @Query("SELECT DISTINCT name FROM exercises")
    fun getAllUniqueNames(): LiveData<List<String>>

    @Query("SELECT id from  exercises WHERE name = :name AND workout_id = :workoutId")
    fun getId(name: String, workoutId: Long): Long

    @Query("UPDATE exercises SET weight = :weight WHERE id = :id")
    fun updateWeight(weight: Float, id: Long)

    @Update
    fun update(vararg exercise: Exercise)

    @Delete
    fun delete(vararg exercises: Exercise)
}
